addons/mail/wizard/mail_compose_message.py

```diff
@@ -516,17 +516,23 @@ def _process_state(self, mail_values_dict):
 
         return mail_values_dict
 
-    def _get_blacklist_record_ids(self, mail_values_dict):
+    def _get_blacklist_record_ids(self, mail_values_dict, recipients_info):
         blacklisted_rec_ids = set()
-        if self.composition_mode == 'mass_mail' and issubclass(type(self.env[self.model]), self.pool['mail.thread.blacklist']):
+        if self.composition_mode == 'mass_mail':
             self.env['mail.blacklist'].flush(['email'])
             self._cr.execute("SELECT email FROM mail_blacklist WHERE active=true")
             blacklist = {x[0] for x in self._cr.fetchall()}
             if blacklist:
```

Comment:
Can't we say: "if not blacklist: return blacklisted_rec_ids", and then gain one indentation level ? That way code stays left-y :)

Reply:
Yes, I have updated it, thanks.